src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs

```diff
@@ -422,25 +422,25 @@ IEnumerable<Config> GetAotConfigs ()
 						aotOptions.Add ("profile-only");
 						foreach (var p in Profiles) {
 							var fp = Path.GetFullPath (p.ItemSpec);
-							aotOptions.Add ($"profile={GetShortPath (fp)}");
+							aotOptions.Add ($"profile={fp}");
 						}
 					}
 					if (!string.IsNullOrEmpty (AotAdditionalArguments))
 						aotOptions.Add (AotAdditionalArguments);
 					if (sequencePointsMode == SequencePointsMode.Offline)
-						aotOptions.Add ("msym-dir=" + GetShortPath (outdir));
+						aotOptions.Add ($"msym-dir={outdir}");
 					if (AotMode != AotMode.Normal)
 						aotOptions.Add (AotMode.ToString ().ToLowerInvariant ());
 
-					aotOptions.Add ("outfile="     + GetShortPath (outputFile));
+					aotOptions.Add ($"outfile={outputFile}"); // no quotes on mac
 					aotOptions.Add ("asmwriter");
 					aotOptions.Add ("mtriple="     + mtriple);
-					aotOptions.Add ("tool-prefix=" + GetShortPath (toolPrefix));
+					aotOptions.Add ($"tool-prefix={toolPrefix}"); // no quotes on mac
 					aotOptions.Add ("ld-flags="    + ldFlags);
-					aotOptions.Add ("llvm-path="   + GetShortPath (sdkBinDirectory));
-					aotOptions.Add ("temp-path="   + GetShortPath (tempDir));
+					aotOptions.Add ($"llvm-path={sdkBinDirectory}");
+					aotOptions.Add ($"temp-path={tempDir}"); // no quotes on mac
 
-					string aotOptionsStr = (EnableLLVM ? "--llvm " : "") + "--aot=" + string.Join (",", aotOptions);
+					string aotOptionsStr = (EnableLLVM ? "--llvm " : "") + "\"--aot=" + string.Join (",", aotOptions) + "\"";
```

Comment:
This is the important line. All of the options inside the `--aot=` argument do not need quotes as long as the entire argument is wrapped in quotes. e.g

```
"--aot=foo=bar bar,doh,cat=true"
```